src/runtimes/runners/TaskCommandRunnerFactory.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { CommandResponse, CommandResponseLike, CommandRunner, ICommandRunnerFactory, normalizeCommandResponseLike } from '@microsoft/container-runtimes';
+
+interface TaskCommandRunnerOptions {
+    taskName: string;
+    workspaceFolder?: vscode.WorkspaceFolder;
+    cwd?: string;
+    alwaysRunNew?: boolean;
+    rejectOnError?: boolean;
+    focus?: boolean;
+    env?: never; // Environment is not needed and should not be used, because VSCode adds it already (due to using `ExtensionContext.environmentVariableCollection`)
+}
+
+export class TaskCommandRunnerFactory implements ICommandRunnerFactory {
+    public constructor(private readonly options: TaskCommandRunnerOptions) {
+    }
+
+    public getCommandRunner(): CommandRunner {

```

Comment:
I am wondering if `getCommandRunner` should be just a factory method, taking options as a parameter. The code consuming it could be simpler then (no need to create a separate "factory" object).

Reply:
It was originally implemented that way, but I cannot remember why we changed it. @danegsta do you remember?

Reply:
If I remember correctly, it was a combination of using a factory object making the typings much more straightforward and making it easier to pass around/re-use a given factory multiple times in a given UI operation as the config should only need to be set once.

Reply:
Not sure about the typing part, but for the re-use, given that the factory's only function is to hold onto options, one could just create the "options" object and re-use it as needed. But in majority of use cases I have seen so far during the review, there is only a single call to `getCommandRunner`, and so the options could be created inline, making the code simpler.

Just sayin ☺️ No biggie, I will let you and Brandon decide what to do.